packages/core/src/classes/transformer/document-client-batch-transformer.ts

```diff
@@ -299,12 +303,14 @@ export class DocumentClientBatchTransformer extends LowOrderTransformers {
 
     return batchItems.reduce(
       (acc, batchItem) => {
-        // is create
-        if (isBatchAddCreateItem(batchItem)) {
+        // is create or put
+        if (isBatchAddCreateItem(batchItem) || isBatchAddPutItem(batchItem)) {
           // transform put item
           const dynamoPutItem = this.toDynamoPutItem(
-            batchItem.create.item,
-            undefined,
+            isBatchAddPutItem(batchItem)
+              ? batchItem.put.item
+              : batchItem.create.item,
+            {overwriteIfExists: isBatchAddPutItem(batchItem)},
```

Comment:
If the only difference between put and create is the value for `overwriteIfExists` option, can we instead remove the WriteBatchPut and add an option in `WriteBatchCreate` only to allow specifying `overwriteIfExists`.

This would require minimum change but will achieve the same.

Reply:
Done!